src/vaadin-dialog-layout.html

```diff
@@ -132,14 +154,34 @@
             };
           }
 
+          ready() {
+            super.ready();
+            this._dialogOpenedChangedListener = this._dialogOpenedChangedListener.bind(this);
+            this.$.dialog.addEventListener('opened-changed', this._dialogOpenedChangedListener);
+          }
+
+          _dialogOpenedChangedListener() {
+            this.opened = this.$.dialog.opened;
+          }
+
           _openedChanged(opened) {
             if (this.opened) {
-              Array.from(this.childNodes).forEach(c => this.$.dialog.$.overlay.$.content.appendChild(c));
-              // Wait to set label until slotted element have been moved.
-              setTimeout(() =>
-                this.__ariaLabel = Array.from(this.$.dialog.$.overlay.$.content.querySelectorAll('[slot=header]'))
-                  .reduce((prev, ele) => prev + ' ' + ele.textContent, '').trim());
+              if (this.editorPosition === '') {
+                Array.from(this.$.editor.childNodes).forEach(c => this.$.dialog.$.overlay.$.content.shadowRoot.appendChild(c));
+                Array.from(this.childNodes).forEach(c => this.$.dialog.$.overlay.$.content.appendChild(c));
+                // Wait to set label until slotted element have been moved.
+                setTimeout(() =>
+                  this.__ariaLabel = Array.from(this.$.dialog.$.overlay.$.content.querySelectorAll('[slot=header]'))
+                    .reduce((prev, ele) => prev + ' ' + ele.textContent, '').trim());
+              }
             }
+
+            // Make sure to reset scroll position
+            this.$.scroller.scrollTop = 0;
+          }
+
+          _computeEditorOpened(opened, editor) {
```

Comment:
Nit: the second param can be changed to smth like `editorPosition` :)
Not a blocker.